flytekit/deck/deck.py

```diff
@@ -10,6 +11,18 @@
 DECK_FILE_NAME = "deck.html"
 
 
+class DeckFields(str, enum.Enum):
```

Comment:
Nit: I think enums in this context is singular:

```suggestion
class DeckField(str, enum.Enum):
```

(Many places needs to be adjusted with this change)